### PR TITLE
I have added all the routes for missing pages

### PR DESCRIPTION
**Pull request by JyothiNandyala**
# Description

I have added all the routes for the missing pages in app.js file.

Fixes # (issue)
#118 
# How to test?

give the command like npm run dev then check all the pages with /(link name).

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by dpfernandes — CHANGES_REQUESTED** [review]
A few inconsistent names here and few other things: 

Delete these

->  <SignIn exact path="/sign-in" />
->  <SignUp exact path="/sign-up" />
->  <ResetPassword exact path="/reset-password" />

Please hyphens on paths when you use 2 words, example: **quiz-results** instead of **quizresults** 
And remove **page** word from the paths

**Comment by Enqira** [comment]
Well done Jyothi, but shouldn't you use   <AuthenticatedRoute exact path="/path-name"> instead of  <Route exact path="/path-name"> for the protected routes, because for instance we don't want the client to be able to access career-page, results profile-page, elevator-pitch, etc if they are not authentificated?

File: src/client/App.js [hunk 1]
```diff
@@ -30,6 +41,36 @@ function App() {
         <SignIn exact path="/sign-in" />
         <SignUp exact path="/sign-up" />
         <ResetPassword exact path="/reset-password" />
+        <Route exact path="/error">
+          <Error404Page />
+        </Route>
+        <Route exact path="/careerpage">
+          <CareerPage />
+        </Route>
+        <Route exact path="/elevatorpitch">
```
Comment (valmdz) [inline src/client/App.js:50]:
```suggestion
        <Route exact path="/elevator-pitch">
```

File: src/client/App.js [hunk 2]
```diff
@@ -30,6 +41,36 @@ function App() {
         <SignIn exact path="/sign-in" />
         <SignUp exact path="/sign-up" />
         <ResetPassword exact path="/reset-password" />
+        <Route exact path="/error">
```
Comment (valmdz) [inline src/client/App.js:44]: The 404 page will be triggered just if the user types the word error on the path, so you probably want to approach this in a different way. I am sure you will be able to find a solution in google :)

File: src/client/App.js [hunk 3]
```diff
@@ -30,6 +41,36 @@ function App() {
...
+        </Route>
+        <Route exact path="/quizresults">
+          <QuizResultPage />
+        </Route>
+        <Route exact path="/registrationpage">
+          <RegistrationPage />
+        </Route>
 
         {/* All routes below are authenticated routes - a user must login first */}
         <AuthenticatedRoute exact path="/profile">
```
[the remark this answers is not in the document]
Comment (JyothiNandyala) [reply src/client/App.js:76]: Yes valentina your right,but still we dont have any authentication for the routers may be we will change those things in the future.

File: src/client/App.js (hunk 1 above)
Comment (JyothiNandyala) [reply src/client/App.js:50]: yes i made the change

**Review by dpfernandes — APPROVED** [review]
👍

**Pull request merged**